quic/transport/ngtcp2/native/streams.nim

```diff
@@ -9,26 +9,26 @@ import chronicles
 logScope:
   topics = "native stream"
 
-proc newStream(connection: Ngtcp2Connection, id: int64): Stream =
-  let stream = newStream(id, newOpenStreamState(connection))
-  connection.setStreamUserData(id, unsafeAddr stream[])
-  return stream
-
 proc openStream*(
     connection: Ngtcp2Connection, unidirectional: bool
 ): Stream {.raises: [QuicError].} =
-  var id: int64
-  if unidirectional:
-    id = connection.openUniStream()
-  else:
-    id = connection.openBidiStream()
-  newStream(connection, id)
+  let stream = newStream(newOpenStreamState(connection))
+  let id =
+    if unidirectional:
+      connection.openUniStream(addr stream[])
+    else:
+      connection.openBidiStream(addr stream[])
+  stream.id = id
+  return stream
```

Comment:
Not the scope of this PR, but could the stream direction be specified in the creation function (i.e. `newStream`)?